src/status_im2/contexts/chat/home/chat_list_item/view.cljs

```diff
@@ -17,61 +20,112 @@
-       :length     size})))
-
-(defn add-parsed-to-subheader
-  [acc {:keys [type destination literal children]}]
-  (let [result (case type
-                 "paragraph"
-                 (reduce
-                  (fn [{:keys [_ length] :as acc-paragraph} parsed-child]
-                    (if (>= length max-subheader-length)
-                      (reduced acc-paragraph)
-                      (add-parsed-to-subheader acc-paragraph parsed-child)))
-                  {:components [quo/text]
-                   :length     0}
-                  children)
-
-                 "mention"
-                 {:components [quo/text (rf/sub [:messages/resolve-mention literal])]
-                  :length     4} ;; we can't predict name length so take the
-                                 ;; smallest possible
-
-                 "status-tag"
-                 (truncate-literal (str "#" literal))
-
-                 "link"
-                 (truncate-literal destination)
-
-                 (truncate-literal literal))]
-    {:components (conj (:components acc) (:components result))
-     :length     (+ (:length acc) (:length result))}))
-
-(defn render-subheader
-  "Render the preview of a last message to a maximum of max-subheader-length characters"
+
+(defn parsed-text-to-one-line
   [parsed-text]
-  (let [result
-        (reduce
-         (fn [{:keys [_ length] :as acc-text} new-text-chunk]
-           (if (>= length max-subheader-length)
-             (reduced acc-text)
-             (add-parsed-to-subheader acc-text new-text-chunk)))
-         {:components [quo/text
-                       {:size                :paragraph-2
-                        :style               {:color (colors/theme-colors colors/neutral-50
-                                                                          colors/neutral-40)
-                                              :width "90%"}
-                        :number-of-lines     1
-                        :ellipsize-mode      :tail
-                        :accessibility-label :chat-message-text}]
-          :length     0}
-         parsed-text)]
-    (:components result)))
+  (reduce
+   (fn [acc {:keys [type literal children destination]}]
+     (case type
+       "paragraph"
+       (str acc (parsed-text-to-one-line children) " ")
+
+       "mention"
+       (str acc (rf/sub [:messages/resolve-mention literal]))
+
+       "status-tag"
+       (str acc literal)
+
+       "link"
+       (str acc destination)
+
+       (str acc literal)))
+   ""
+   parsed-text))
+
+(defn extract-text-from-message
+  [{:keys [content]}]
+  (let [{:keys [parsed-text text]} content]
+    (if parsed-text (parsed-text-to-one-line parsed-text) text)))
+
+
+(defn last-message-preview
+  "Render the preview of a last message to a maximum of max-subheader-length characters"
+  [chat-id _]
+  (reagent/create-class
+   {:component-did-mount
+    (fn [_]
+      (rf/dispatch [:pin-message/load-pin-messages chat-id]))
```

Comment:
could you pls elaborate on why we want this here and only once when mount?

Reply:
We need the details of the pinned message to extract text for message preview. But they are loaded only when chat opened, so I've added this call, to get them earlier. And I wanted to reduce the number of calls in case of re-rendering, but after your question I think maybe it is worth calling on every re-render because it can be caused because the last message changed. Wdyt?

Reply:
@vkjr, calling once seems alright to me if the last message will change anyway due to signals emitted by status-go, or isn't it the case?

Reply:
@ilmotta, appears that it is better to be called on render because while the messages list is updated by signal, the pinned messages don't and that causes issue.

@flexsurfer, agree, let me check if I'll be able to pass the whole last message along with the chat preview from status-go instead of separate fields.

Reply:
we've changed this because of performance, we don't want entire message, sometimes its too heavy, so we just need to add fileds we need